api/app/controllers/folders_controller.rb

```diff
@@ -0,0 +1,66 @@
+class FoldersController < ApplicationController
+  # before_action :authenticate_user!, except: [:index, :show]
+  before_action :authorize_user!, only: [ :update, :destroy ]
+
+
+  # GET /folders
+  def index
+    folders = Folder.all
+    render json: FolderSerializer.new(folders).serializable_hash.to_json
+  end
+
+  # GET /my_folders
+  def my_folders
+    authenticate_user!
```

Comment:
Even though it would be only be `only: [ :my_folders ]` I think I would still put this call to authenticate_user! into a before_action, just so the endpoints themselves are strictly about what they do and now the barriers around doing them